types/index.d.ts

```diff
@@ -210,8 +210,8 @@ declare module "miragejs/server" {
   type MaybePromise<T> = T | PromiseLike<T>;
 
   /** A callback that will be invoked when a given Mirage route is hit. */
-  export type RouteHandler<Registry> = (
-    schema: Schema<Registry>,
+  export type RouteHandler<T extends Registry<any, any>> = (
```

Comment:
I don't think this is what we want to do. Instead, I think Registry should be given defaults:
```ts
declare module "miragejs/-types" {
  ⋮
  export type Registry<
    Models extends Record<string, ModelDefinition> = Record<string, ModelDefinition>,
    Factories extends Record<string, FactoryDefinition> = Record<string, FactoryDefinition>
  > = {
    [K in keyof Models | keyof Factories]: ModelInstance<
      ExtractModelData<Models, K> & ExtractFactoryData<Factories, K>
    >;
  };
  ⋮
```
and then just have
```suggestion
  export type RouteHandler<T extends Registry> = (
```

Reply:
@jamescdavis Fantastic suggestion. Thank you. It really makes sense and clean up the the usage of Registry type. Love it. Please review the update. What do you think? :)